internal/storage/storage.go

```diff
@@ -138,13 +139,18 @@ func (s *ConfigMaps) buildObject(obj client.Object, owner client.Object) (*corev
 	}
 	gvk := obj.GetObjectKind().GroupVersionKind()
 
+	objName := obj.GetName()
+	if strings.Contains(objName, ":") {
+		objName = strings.ReplaceAll(objName, ":", "-")
+	}
+
 	labels := map[string]string{
 		"core.rukpak.io/owner-name":       owner.GetName(),
```

Comment:
We currently have "core.rukpak.io/owner-name" used in both bundle storage and bundle instance installation, which could result in some incorrect assumptions if a bundle and bundle instance have the same metadata.name.

I think we need to separate these labels (e.g. `core.rukpak.io/bundle-name` and `core.rukpak.io/bundle-instance-name`)

Reply:
I'm fine with updating to differentiate between bundle and bundleinstance controller resources, but the only issue I see when poking at this code locally is how to handle the manager's cache if we have two different label selecting mechanisms now. I didn't see anything in the `labels` package for exposing a label selector for objects that have the core.rukpak.io/bundle-name label OR the core.rukpak.io/bundleinstance-name label present.

Reply:
I think I'm leaning towards handling this in a follow-up in either case as this PR fixes a couple of things at this point. Thoughts?

Reply:
Follow-up is fine. These labels would be used on separate types, so we should be able to use different KV pairs for `SelectorsByObject` in the `cache.Options`.

Reply:
Alright sounds good to me - I'll track this locally and try to tackle this tomorrow.

Reply:
Actually, you're totally right. This isn't a selector on `Bundle` and `BundleInstance` objects. This is a selector on the rest of the objects. And I think that's the single `dependentSelector`. Hmm...

Reply:
Added a new `core.rukpak.io/owner-kind` label. And an `In` selector on `Bundle` or `BundleInstance` to find objects with either value. https://github.com/operator-framework/rukpak/pull/76